driver-legacy/src/main/com/mongodb/MongoClient.java

```diff
@@ -21,107 +21,104 @@
+ * list of replica set members or a list of mongos servers.</p>
  *
  * <p>By default, all read and write operations will be made on the primary, but it's possible to read from secondaries by changing the read
  * preference:</p>
  * <pre>
- * mongoClient.setReadPreference(ReadPreference.secondaryPreferred());
+ * new MongoClient("mongodb://localhost:27017,localhost:27018,localhost:27019?readPreference=primary")
+ * new MongoClient(MongoClientSettings.builder()
+ *   .applyConnectionString("mongodb://localhost:27017,localhost:27018,localhost:27019/?readPreference=primary")
+ *   .build())
+ * new MongoClient(MongoClientSettings.builder()
+ *   .applyConnectionString("mongodb://localhost:27017,localhost:27018,localhost:27019")
+ *   .readPreference(ReadPreference.primary())
+ *   .build())
  * </pre>
  * <p>By default, all write operations will wait for acknowledgment by the server, as the default write concern is {@code
- * WriteConcern.ACKNOWLEDGED}.</p>
- *
- * <p>Note: This class supersedes the {@code Mongo} class.  While it extends {@code Mongo}, it differs from it in that the default write
- * concern is to wait for acknowledgment from the server of all write operations.  In addition, its constructors accept instances of {@code
- * MongoClientOptions} and {@code MongoClientURI}, which both also set the same default write concern.</p>
- *
- * <p>In general, users of this class will pick up all of the default options specified in {@code MongoClientOptions}.
+ * WriteConcern.ACKNOWLEDGED}.  It's possible to change this with a setting:</p>
+ * <pre>
+ * new MongoClient("mongodb://localhost:27017,localhost:27018,localhost:27019?w=majority")
+ * new MongoClient(MongoClientSettings.builder()
+ *   .applyConnectionString("mongodb://localhost:27017,localhost:27018,localhost:27019/?w=majority")
+ *   .build())
+ * new MongoClient(MongoClientSettings.builder()
+ *   .applyConnectionString("mongodb://localhost:27017,localhost:27018,localhost:27019")
+ *   .writeConcern(WriteConcern.MAJORITY)
+ *   .build())
+ * </pre>
+ * <p>In general, users of this class will pick up all of the default options specified in {@code MongoClientSettings}.
  *
- * @see ReadPreference#primary()
- * @see com.mongodb.WriteConcern#ACKNOWLEDGED
- * @see MongoClientOptions
- * @see MongoClientURI
+ * @see ConnectionString
+ * @see MongoClientSettings
  * @since 2.10.0
  */
 public class MongoClient implements Closeable {
 
     private final ConcurrentMap<String, DB> dbCache = new ConcurrentHashMap<>();
 
     private final MongoClientOptions options;
-    private final MongoCredential credential;
 
     private final BufferProvider bufferProvider = new PowerOfTwoBufferPool();
 
     private final ConcurrentLinkedQueue<ServerCursorAndNamespace> orphanedCursors = new ConcurrentLinkedQueue<>();
     private final ExecutorService cursorCleaningService;
-    private final MongoClientDelegate delegate;
+    private final MongoClientImpl delegate;
```

Comment:
This allows a major simplification, since we can just delegate everything very simply now.